component/prometheus/remotewrite/remote_write.go

```diff
@@ -71,7 +70,10 @@ func NewComponent(o component.Options, c Arguments) (*Component, error) {
 		remoteStore: remoteStore,
 		storage:     storage.NewFanout(o.Logger, walStorage, remoteStore),
 	}
-	res.receiver = &prometheus.Receiver{Receive: res.Receive}
+	res.receiver = &appendable{
+		inner:       res.storage,
+		componentID: o.ID,
+	}
```

Comment:
If we make the interceptor its own storage.Appendable, then you should be able to replace the custom appendable struct here with the interceptor to reduce some code, I think.

Reply:
I was torn on this one and did it both ways, and felt that remote_write having its own custom one was warranted because anything interfacing with remote_write will have to do the translation ie appendexemplar and updatemetadata. Where my gut is that other components wont have any customer behavior on those.